src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1019,40 +1019,27 @@ namespace AirLoopHVACDOAS {
     void AirLoopDOAS::GetDesignDayConditions(EnergyPlusData &state)
     {
 
-        int constexpr summerDesignDayTypeIndex(9);
-        int constexpr winterDesignDayTypeIndex(10);
-
-        for (size_t i = 1; i <= state.dataWeatherManager->DesDayInput.size(); i++) {
-            auto &desDayInput = state.dataWeatherManager->DesDayInput(i);
-            // Summer design day
-            if (desDayInput.DayType == summerDesignDayTypeIndex) {
-                // keep design day humiditiy ratio from the same peak temperature design day (i.e., don't use a humrat from a different day)
-                if (desDayInput.MaxDryBulb > this->SizingCoolOATemp) {
-                    this->SizingCoolOATemp = desDayInput.MaxDryBulb;
-                    if (desDayInput.HumIndType == WeatherManager::DDHumIndType::WetBulb) { // wet bulb temperature
-                        this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdbTwbPb(
-                            state, this->SizingCoolOATemp, desDayInput.HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                    } else if (desDayInput.HumIndType == WeatherManager::DDHumIndType::DewPoint) { // dewpoint
-                        this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(state, desDayInput.HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                    } else if (desDayInput.HumIndType == WeatherManager::DDHumIndType::HumRatio) {
-                        this->SizingCoolOAHumRat = desDayInput.HumIndValue;
-                    } // else { // What about other cases?
+        // int constexpr summerDesignDayTypeIndex(9);
+        // int constexpr winterDesignDayTypeIndex(10);
+        for (size_t env = 1; env <= state.dataWeatherManager->Environment.size(); ++env) {
+            switch (state.dataWeatherManager->Environment(env).KindOfEnvrn) {
```

Comment:
Given that this is only acting on two possible conditions, I'm not _convinced_ that the switch block is better, but at the same time, I personally think it could add a lot of readability with a couple tweaks.  I'm curious what @amirroth would comment here.

Right now the code looks like:
```
// int constexpr summerDesignDayTypeIndex(9);
// int constexpr winterDesignDayTypeIndex(10);
for (size_t env = 1; env <= state.dataWeatherManager->Environment.size(); ++env) {
  switch (state.dataWeatherManager->Environment(env).KindOfEnvrn) {
  case DataGlobalConstants::KindOfSim::DesignDay:
  case DataGlobalConstants::KindOfSim::RunPeriodDesign: {
    if (state.dataWeatherManager->Environment(env).maxCoolingOATSizing >
        this->SizingCoolOATemp) {
      this->SizingCoolOATemp = state.dataWeatherManager->Environment(env).maxCoolingOATSizing;
      this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(
          state,
          state.dataWeatherManager->Environment(env).maxCoolingOADPSizing,
          DataEnvironment::StdPressureSeaLevel);
    }
    if (state.dataWeatherManager->Environment(env).minHeatingOATSizing <
        this->HeatOutTemp) {
      this->HeatOutTemp = state.dataWeatherManager->Environment(env).minHeatingOATSizing;
      this->HeatOutHumRat = Psychrometrics::PsyWFnTdpPb(
          state,
          state.dataWeatherManager->Environment(env).minHeatingOADPSizing,
          DataEnvironment::StdPressureSeaLevel);
    }
  } break;
  case DataGlobalConstants::KindOfSim::RunPeriodWeather: {
  } break;
  default: {
  } break;
  }
}
```

I think this actually looks easiest to read as a range based for loop with a single IF condition, like::

```
for (auto &env : state.dataWeatherManager->Environment) {
  if (env.KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay || env.KindOfEnvrn == DataGlobalConstants::KindOfSim::RunPeriodDesign) {
    if (env.maxCoolingOATSizing > this->SizingCoolOATemp) {
      this->SizingCoolOATemp = env.maxCoolingOATSizing;
      this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(state, env.maxCoolingOADPSizing, DataEnvironment::StdPressureSeaLevel);
    }
    if (env.minHeatingOATSizing < this->HeatOutTemp) {
      this->HeatOutTemp = env.minHeatingOATSizing;
      this->HeatOutHumRat = Psychrometrics::PsyWFnTdpPb(state, env.minHeatingOADPSizing, DataEnvironment::StdPressureSeaLevel);
    }
  }
}
```

The range-based for loop ensures we only ever loop across the size of the vector, we won't go out of bounds.  In some cases in E+ we obviously don't want to do that, so plain for loops are great.  And like in the other example I comments, the single index is used in multiple arrays in the loop, so that needs to be a regular for loop.  But for cases where you are only traversing a single vector from start to finish, the range-based loops make the code super readable, you don't have to manually create a reference in the loop, and help avoid possible bugs later.

Reply:
`If/else` is good for two cases. One tweak I would make is:

```cpp
if (env.kindOfEnvrn != KindOfSim::DesignDay && env.kindOfEnvrn != KindOfSim::RunPeriodDesign) continue;
```

It reduces the level of logic nesting.  

PS Don't get me starting on naming something `KindOf`. Couldn't you at least have named it `Kinda`?